### ISSUE TITLE
Wishlist for improving testing/demoing

### ISSUE THREAD

**Issue by maelle** [cynkra/fledge#714]
- more use of a local remote.
- also use this local remote in a demo in an article.
- mock the CRAN release somehow, for testing and also for demoing.
- there's a tension between testing and debugging tests regarding the use of `local_` functions in tests: changing the local directory makes it hard, well clumsy, to run tests interactively.
- #702 
- #695 
- #8 
- #103

@krlmlr

**Comment by maelle** [comment]
For `create_pull_request()`, use test switch or mocking.

**Comment by krlmlr** [comment]
Let's start with a simple R script that goes through the entire release process. It should work offline, for speed, mocking away anything that might need connectivity.

**Comment by krlmlr** [comment]
Seems good enough now?

**Comment by maelle** [comment]
yes but we don't have "mock the CRAN release somehow, for testing and also for demoing." in particular

**Comment by krlmlr** [comment]
We're mocking it away, we can call `release()` in the tests without failing. What else do we need to do?

**Comment by maelle** [comment]
right, we're avoiding more than mocking, but i guess mocking wouldn't bring much
